src/components/lyne-text-input/lyne-text-input.tsx

```diff
@@ -0,0 +1,300 @@
+      this._additionalInputClasses.push('input-wrapper--error');
+    }
+
+  }
+
+  private _getAdditionalInputAttributes(): void {
+
+    this._addtitionalInputAttributes = {};
+
+    if (this.inputDisabled) {
+      this._addtitionalInputAttributes = {
+        ...this._addtitionalInputAttributes,
+        disabled: true
+      };
+    }
+
+    if (this.inputPattern) {
+      this._addtitionalInputAttributes = {
+        ...this._addtitionalInputAttributes,
+        pattern: this.inputPattern
+      };
+    }
+
+    if (this.inputMaxLength) {
+      this._addtitionalInputAttributes = {
+        ...this._addtitionalInputAttributes,
+        maxlength: this.inputMaxLength
+      };
+    }
+
+    if (this.inputMinLength) {
+      this._addtitionalInputAttributes = {
+        ...this._addtitionalInputAttributes,
+        minlength: this.inputMinLength
+      };
+    }
+
+    if (this.inputError) {
+      this._addtitionalInputAttributes = {
+        ...this._addtitionalInputAttributes,
+        'aria-invalid': 'true'
+      };
+    }
+
+  }
+
+  /**
+   * TODO
+   * modify lyne-button to allow for icon toggle
+   * and add toggle handling in here.
+   */
+  private _registerShowPasswordToggle(): void {
+
+    /**
+     * if (this.inputAutoCompleteValue !== 'new-password') {
+     *  return;
+     *}
+     */
+
+  }
+
+  /**
+   * This will improve the announcement of the label
+   * in Voice Over on iOS Safari
+   */
+  private _prepareAriaLabelOfLabel(): void {
+
+    if (this.inputRequired) {
+      this._labelAriaLabel = this.label;
+    } else {
+      this._labelAriaLabel = `${this.label} ${i18nOptional[this._currentLanguage]}.`;
+    }
+
+    if (this.inputError) {
+      this._labelAriaLabel += `. ${i18nMandatoryField[this._currentLanguage]}`;
+    }
+  }
+
+  public componentWillLoad(): void {
+    if (this.inputId) {
+      this._id = this.inputId;
+    } else {
+      this._id = `input-${guid()}`;
+    }
+  }
+
+  public render(): JSX.Element {
+
+    this._getAdditionalStyleClasses();
+    this._getAdditionalInputAttributes();
+    this._registerShowPasswordToggle();
+    this._prepareAriaLabelOfLabel();
+
+    /**
+     * Adding the aria-hidden attributes here might
+     * seem to be a very wrong choice. Since we are
+     * including the error message inside of the
+     * aria-label value of the label element though,
+     * we need to hide some elements from assistive
+     * technology here. By doing so, we can improve
+     * the user flow through the form, especially
+     * on Voice Over on iOS which would otherwise
+     * stop at and read all elements individually.
+     * This was tested in various Desktop Browsers
+     * with JAWS and NVDA as well.
+     */
+
+    return (
+      <div
+        class={`input-wrapper ${this._additionalInputClasses.join(' ')}`}
+      >
+        <div class='input-wrapper__inner'>
+          {this.icon
+            ? <span class='link__icon'><slot name='icon'/></span>
```

Comment:
`link` in this class name doesn't make sense to me

Reply:
Sure, total nonsene from my side. Fixed.